### ISSUE TITLE
binary:logistic most predict result very close to 0

### ISSUE THREAD

**Issue by readilychen** [dmlc/xgboost#4479]
1.with our recently built models, predict result are very close to 0, such as below:
5.88484E-06
3.42964E-06
3.42176E-06
9.57E-07
8.68E-07
3.90E-07
1.89E-07
9.22E-08
7.62E-08

2. previous model we could get normal results(mainly between 0.1 and 1), such as:
0.999991565
0.999515528
0.998715854
9.80E-01
9.78E-01
9.75E-01
9.57E-01
9.36E-01
9.04E-01
9.03E-01
8.18E-01
8.15E-01
7.71E-01
7.58E-01
7.05E-01
6.84E-01
6.45E-01
6.43E-01
4.95E-01
4.78E-01
4.73E-01
4.33E-01
4.20E-01
4.04E-01

No train params changed beteew the two versions. only train data are different --- we collect data  by date,  and the two train data scale not changed much. 
why this happens? Is any one encountered this before?

**Comment by trivialfis** [comment]
@readilychen 
- Could you run the same dataset with two different versions?
- If above result is abnormal, could you provide us the git commit hashes (or release version) of two different versions?  Also with a reproducible script so that we can look into it?

**Comment by readilychen** [comment]
I trained a new model(compressed in attch c_20190425000520.zip) using the previous dataset, but the results  still very closed to 0.

test features could be as bellow(more date attched in file "features.txt"):
     2.1806559509808634E-7,1603076700,0:4233.0,1:103.0,8:44.0,9:1916.0,10:5.0,11:1.0,17:1.0,25:1.0,31:1.0,33:0.57725316,34:0.99894726,35:0.08380299,36:0.093063265,38:0.9155823,39:0.15018855,40:0.32405087,42:0.9645653,45:0.47988382,48:0.14121655,50:0.04896082,52:0.17307349,53:0.124353,54:0.10988779,56:0.3147719,58:1.0,59:0.47988382,66:1.0,68:1.0,
3.915927776246853E-4,1608313900,0:4242.0,1:233.0,8:14.0,10:10.0,11:1.0,15:1.0,25:1.0,31:1.0,33:0.57725316,34:0.99894726,35:0.08380299,36:0.093063265,38:0.9155823,39:0.15018855,40:0.32405087,42:0.9645653,45:0.47988382,48:0.14121655,50:0.04896082,52:0.17307349,53:0.124353,54:0.10988779,56:0.3147719,66:1.0,68:1.0,
0.0014463282881032665,1600994800,0:4287.0,1:168.0,8:22.0,9:2551.0,10:9.0,11:2.0,12:1.0,14:1.0,17:1.0,25:1.0,31:1.0,33:0.57725316,34:0.99894726,35:0.08380299,36:0.093063265,38:0.9155823,39:0.15018855,40:0.32405087,42:0.9645653,45:0.47988382,48:0.14121655,50:0.04896082,52:0.17307349,53:0.124353,54:0.10988779,56:0.3147719,59:0.47988382,66:1.0,68:1.0

while the first column is the predict result,  the others are input features.


my training code is as follow:

`
val paramMap = List(
			"verbosity" -> 2,
			"eta" -> eta,
			"max_depth" -> 6,
			"objective" -> "binary:logistic",
			"gamma" -> 0.0,
			"min_child_weight" -> 0.01,
			"subsample" -> 0.8,
			"colsample_bytree" -> 0.8,
			"scale_pos_weight" -> 25,
			"num_round" -> 1000,
			"num_workers" -> 20,
			"nthread" -> 4
		).toMap
		//
		val classifier = new XGBoostClassifier("c", paramMap)
		classifier.setEvalMetric("auc")
		classifier.setFeaturesCol("features")
		classifier.setLabelCol("label")
		//    classifier.setTrainTestRatio(0.8)
		classifier.setEvalSets(Map("eval" -> test))
		//
		val xgboostModel = classifier.fit(train_data)

`



[features.txt](https://github.com/dmlc/xgboost/files/3200407/features.txt)



[c_20190425000520.zip](https://github.com/dmlc/xgboost/files/3200374/c_20190425000520.zip)

**Comment by Denisevi4** [comment]
whats eta value here?

**Comment by readilychen** [comment]
> whats eta value here?

eta = 0.8

**Comment by readilychen** [comment]
I traced our online predict process, then find out that 2 features lossed online, which leading this problem.
